src/CairoLib.sol

```diff
@@ -28,25 +28,20 @@ library CairoLib {
         return result;
     }
 
-    function callCairo(CairoCall memory call)
-        internal
-        returns (bytes memory)
-    {
+    function callCairo(CairoCall calldata call) internal returns (bytes memory) {
         return callCairo(call.contractAddress, call.functionSelector, call.data);
     }
 
     function callCairo(uint256 contractAddress, uint256 functionSelector) internal returns (bytes memory) {
-        uint256[] memory data = new uint256[](0);
-        return callCairo(contractAddress, functionSelector, data);
+        return callCairo(contractAddress, functionSelector);
```

Comment:
I am not sure this can work: it would call itself recursively.
You would need to pass the data as the actual function overloaded has a third parameter.